lighthouse-core/gather/driver.js

```diff
@@ -643,18 +644,18 @@ class Driver {
             } else {
               log.verbose('Driver', `CPU has been idle for ${timeSinceLongTask} ms`);
               const timeToWait = waitForCPUQuiet - timeSinceLongTask;
-              lastTimeout = setTimeout(() => checkForQuiet(driver, resolve), timeToWait);
+              lastTimeout = setTimeout(() => checkForQuiet(driver, resolve, reject), timeToWait);
             }
           }
-        });
+        }).catch(e => reject(e));
     }
 
     /** @type {(() => void)} */
     let cancel = () => {
       throw new Error('_waitForCPUIdle.cancel() called before it was defined');
     };
     const promise = new Promise((resolve, reject) => {
-      checkForQuiet(this, resolve);
+      checkForQuiet(this, resolve, reject);
```

Comment:
```suggestion
      checkForQuiet(this, resolve).catch(reject);
```

Reply:
oh you'd need to change the `return;` to `return Promise.resolve();` too, but I'm not sure I follow what impact the resolve `() => void` is? Maybe I'm skipping a step somewhere :)

Reply:
I was misinterpreting the linting hint as being important, it was lying to me!